lib/browser/api/web-contents.js

```diff
@@ -110,9 +110,15 @@ const webFrameMethodsWithResult = [
 ]
 
 const asyncWebFrameMethods = function (requestId, method, callback, ...args) {
-  this.send('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_METHOD', requestId, method, args)
-  ipcMain.once(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, function (event, result) {
-    if (callback) callback(result)
+  return new Promise((resolve, reject) => {
+    this.send('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_METHOD', requestId, method, args)
+    ipcMain.once(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_${requestId}`, function (event, result) {
+      if (callback) callback(result)
+      resolve(result)
+    })
+    ipcMain.once(`ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_ERROR_${requestId}`, (event, error) => {
```

Comment:
Hmm, so this event listener will hang around forever if the promise is never rejected, and the other one will hang around forever if the promise is rejected right?

Maybe we can use `ELECTRON_INTERNAL_BROWSER_ASYNC_WEB_FRAME_RESPONSE_` for both cases and just have two arguments `(event, result, error)` and reject when error is present and resolve otherwise with result.